src/test/java/org/broadinstitute/hellbender/engine/FeatureSupportIntegrationTest.java

```diff
@@ -83,6 +92,25 @@ public void testFeaturesAsIntervalsUnrecognizedFormatFile() throws IOException {
         testSpec.executeTest("testFeaturesAsIntervals", this);
     }
 
+    @Test
+    public void testSitesOnlyMode() {
+        File out = createTempFile("GTStrippedOutput", "vcf");
+        String[] args = new String[] {
+                "-V",  FEATURE_INTEGRATION_TEST_DIRECTORY + "vcf_with_genotypes.vcf",
+                "--" + StandardArgumentDefinitions.SITES_ONLY_LONG_NAME,
+                "-O",
+                out.getAbsolutePath()};
+        new Main().instanceMain(makeCommandLineArgs(Arrays.asList(args), SelectVariants.class.getSimpleName()));
+
+        // Assert that the genotype field has been stripped from the file
+        Pair<VCFHeader, List<VariantContext>> results = VariantContextTestUtils.readEntireVCFIntoMemory(out.getAbsolutePath());
+
+        Assert.assertFalse(results.getLeft().hasGenotypingData());
+        for (VariantContext v: results.getRight()) {
+            Assert.assertFalse(v.hasGenotypes());
+        }
+    }
+
```

Comment:
Could you also add a similar test to `HaplotypeCallerIntegrationTest` to prove that it honors `--sites-only`, since that is a special case?